static/app/components/profiling/flamegraphZoomView.tsx

```diff
@@ -321,67 +306,40 @@ function FlamegraphZoomView({
     flamegraphSearch,
   ]);
 
-  useEffect(() => {
-    if (!flamegraphCanvas || !flamegraphView || !selectedFrameRenderer) {
-      return undefined;
-    }
+  const selectedFramesRef = useRef<FlamegraphFrame[] | null>(null);
 
-    const drawFocusedFrameBorder = () => {
-      selectedFrameRenderer.draw(
-        focusedRects,
-        {
-          BORDER_COLOR: flamegraphTheme.COLORS.FOCUSED_FRAME_BORDER_COLOR,
-          BORDER_WIDTH: flamegraphTheme.SIZES.FOCUSED_FRAME_BORDER_WIDTH,
-        },
-        flamegraphView.fromConfigView(flamegraphCanvas.physicalSpace)
+  useEffect(() => {
```

Comment:
Should this be a `useEffectAfterFirstRender` so that it doesn't reset `selectedFramesRef` later?

Reply:
I dont think it makes a difference, what we want is to always set the selectedFramesRef if highlightedFrame is set. I'm not 100% happy with how this works tbh, as I mentioned in DM, we start to have a bunch of adjacent state being initialized in different places, I may refactor this soon as it's getting tough to track